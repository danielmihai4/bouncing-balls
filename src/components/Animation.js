import React from 'react';
import Canvas from './Canvas';

export class Animation extends React.Component {
    constructor(props) {
      super(props);
      this.updateAnimationState = this.updateAnimationState.bind(this);
    }
  
    componentDidMount = () => {
      this.rAF = requestAnimationFrame(this.updateAnimationState);
    }
  
    updateAnimationState = () => {
      this.rAF = requestAnimationFrame(this.updateAnimationState);
    }
  
    componentWillUnmount = () => {
      cancelAnimationFrame(this.rAF);
    }
  
    render() {
      return <Canvas />;
    }
  }

export default Animation;