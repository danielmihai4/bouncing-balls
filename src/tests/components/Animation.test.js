import React from 'react';
import { shallow } from 'enzyme';
import Animation from '../../components/Animation';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<Animation />);
});

test('should render Animation correctly', () => {
    expect(wrapper).toMatchSnapshot();
});