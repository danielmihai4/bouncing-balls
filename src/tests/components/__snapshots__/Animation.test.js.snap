// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Animation correctly 1`] = `<Connect(Canvas) />`;
